src/runtime/hexagon/hexagon_device_api.h

```diff
@@ -54,17 +55,26 @@ class HexagonDeviceAPI final : public DeviceAPI {
   void AcquireResources() {
     CHECK_EQ(runtime_hexbuffs, nullptr);
     runtime_hexbuffs = std::make_unique<HexagonBufferManager>();
-    LOG(INFO) << "runtime_hexbuffs created";
+    DLOG(INFO) << "runtime_hexbuffs created";
     mgr = runtime_hexbuffs.get();
+
+    CHECK_EQ(runtime_threads, nullptr);
+    runtime_threads = std::make_unique<HexagonThreadManager>(threads, stack_size, pipe_size);
+    DLOG(INFO) << "runtime_threads created";
   }
 
   //! \brief Ensures all runtime resources are freed
   void ReleaseResources() {
+    CHECK(runtime_threads) << "runtime_threads was not created in AcquireResources";
+    runtime_threads.reset();
```

Comment:
Just to be clear here (since it might not be apparent from the thread manager interface): calling the destructor of the `HexagonThreadManager` will block until all of its threads have finished working. If work has been dispatched but the threads haven't been started yet, the destructor will start the threads and wait for them to finish.

I'm not sure on the use cases we want to cover here, but there is no "graceful crash" functionality right now, in case you wanted `ReleaseResources()` to be capable of terminating running (or deadlocked) threads.

Reply:
Ok, good to know.  I think that's ok for now.  We may want a "force" shutdown option in the future.